internalsdk/pro/pro.go

```diff
@@ -60,40 +59,21 @@ func NewClient(baseURL string, opts *Opts) ProClient {
 	client := &proClient{
 		userConfig: opts.UserConfig,
 	}
-	client.webclient = webclient.NewRESTClient(defaultwebclient.SendToURL(httpClient, baseURL, client.setUserHeaders(), nil))
+	client.webclient = webclient.NewRESTClient(defaultwebclient.SendToURL(httpClient, baseURL, prepareProRequest(opts.UserConfig), nil))
 	return client
 }
 
-func (c *proClient) setUserHeaders() func(client *resty.Client, req *resty.Request) error {
-	return func(client *resty.Client, req *resty.Request) error {
-
-		uc := c.userConfig()
-
+// prepareProRequest normalizes requests to the pro server with device ID, user ID, etc set.
+func prepareProRequest(userConfig func() common.UserConfig) func(client *resty.Client, req *http.Request) error {
+	return func(client *resty.Client, req *http.Request) error {
+		uc := userConfig()
 		req.Header.Set("Referer", "http://localhost:37457/")
```

Comment:
@atavism Why do we need to pass Referer header, Can we remove it?

Reply:
I think that was originally added because a particular Pro server endpoint expected a 'Referer' header to be passed to where to redirect the user based on the app's success/fail states. I think it should be fine to omit it now as that's no longer required on the pro server